lib/whois/parsers/whois.corporatedomains.com.rb

```diff
@@ -22,6 +22,39 @@ class WhoisCorporatedomainsCom < BaseIcannCompliant
       self.scanner = Scanners::BaseIcannCompliant, {
           pattern_available: /^No match for/
       }
+
+      property_supported :created_on do
+        node("Creation Date") do |value|
+          # Update the format so the value is parsed properly
+          value = reformat_slash_date_format(value) if value.include?("/")
+
+          parse_time(value)
+        end
+      end
+
+      property_supported :expires_on do
+        node("Registrar Registration Expiration Date") do |value|
+          # Update the format so the value is parsed properly
+          value = reformat_slash_date_format(value) if value.include?("/")
+
+          parse_time(value)
+        end
+      end
+
+      private
+
+      # There are 2 different date formats which are used for creation and expiration dates:
+      #   - hyphen-separated date format YYYY-MM-DDTHH:MM:SSZ
+      #   - slash-separated date format MM/DD/YYYY HH:MM:SS
+      #
+      # The latter format causes problems because `parse_time` parsed as (DD/MM/YYYY HH:MM:SS)
+      def reformat_slash_date_format(value)
+        date, time = value.split
+        month, day, year = date.split("/")
+        reformatted_date = [year, month, day].join("-")
```

Comment:
`Time.utc(year, month, day)` will give you a Time object which you can then convert to a string with `to_s` or `as_json`. The manual array join and string interpolation is not needed then.

Reply:
If I'm already creating a Time object, it doesn't seem like there's any reason to convert it to a string, add the time, and then pass it in to `parse_time`, is there?

Reply:
Yeah just returning the Time you generate seems good.